### ISSUE TITLE
Login

### ISSUE THREAD

**Issue by zwhitchcox** [LeetCode-OpenSource/vscode-leetcode#556]
## 🐛 Bug Report

Can't login in any of the 4 methods. I have made sure I have the right password and even tried the LEETCODE_SESSION cookie in my browser. Nothing works.

Is this dead? Because if it is, it would be nice if you could let us know!

## To Reproduce

Try to log in with each method

## Extension Output
```
[ERROR] You are not login yet?
login: pass: - Signing in leetcode.com
[ERROR] invalid password?
login: pass: - Signing in leetcode.com
[ERROR] invalid password?
login: pass: [ERROR] Login failed. Please make sure the credential is correct.
login: pass: - Signing in leetcode.com
[ERROR] invalid password?
login: pass: [ERROR] Login failed. Please make sure the credential is correct.
login: login: cookie: login: cookie: c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555
    return cb('invalid cookie?');
           ^

TypeError: cb is not a function
    at parseCookie (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555:12)
    at Plugin.plugin.cookieLogin (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:578:22)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\commands\user.js:119:12
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:336:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\utile\node_modules\async\lib\async.js:154:25
    at assembler (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:333:9)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:342:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:625:5
    at Interface.onLine (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\read\lib\read.js:111:5)
    at Interface.emit (events.js:198:13)
login: cookie: c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555
    return cb('invalid cookie?');
           ^

TypeError: cb is not a function
    at parseCookie (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555:12)
    at Plugin.plugin.cookieLogin (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:578:22)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\commands\user.js:119:12
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:336:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\utile\node_modules\async\lib\async.js:154:25
    at assembler (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:333:9)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:342:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:625:5
    at Interface.onLine (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\read\lib\read.js:111:5)
    at Interface.emit (events.js:198:13)
login: login: cookie: c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555
    return cb('invalid cookie?');
           ^

TypeError: cb is not a function
    at parseCookie (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555:12)
    at Plugin.plugin.cookieLogin (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:578:22)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\commands\user.js:119:12
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:336:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\utile\node_modules\async\lib\async.js:154:25
    at assembler (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:333:9)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:342:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:625:5
    at Interface.onLine (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\read\lib\read.js:111:5)
    at Interface.emit (events.js:198:13)
login: cookie: c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555
    return cb('invalid cookie?');
           ^

TypeError: cb is not a function
    at parseCookie (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555:12)
    at Plugin.plugin.cookieLogin (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:578:22)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\commands\user.js:119:12
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:336:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\utile\node_modules\async\lib\async.js:154:25
    at assembler (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:333:9)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:342:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:625:5
    at Interface.onLine (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\read\lib\read.js:111:5)
    at Interface.emit (events.js:198:13)
login: 
```
Guidance: Press `Ctrl+Shift+U`, and toggle the channel to `LeetCode`.

## Your Environment

- *os*: Windows 10
- *extension settings*: None
- *nodejs version*:10.16.3
- *vscode version*:1.41.1
- *extension version*:0.16.0

**Comment by zwhitchcox** [comment]
Ok, I hadn't tried the LinkedIn method, this is the output when I try that

```
TypeError: cb is not a function
    at parseCookie (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:555:12)
    at Plugin.plugin.cookieLogin (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\plugins\leetcode.js:578:22)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\vsc-leetcode-cli\lib\commands\user.js:119:12
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:336:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\utile\node_modules\async\lib\async.js:154:25
    at assembler (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:333:9)
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:342:32
    at c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\prompt\lib\prompt.js:625:5
    at Interface.onLine (c:\Users\zwhitchcox\.vscode\extensions\shengchen.vscode-leetcode-0.16.0\node_modules\read\lib\read.js:111:5)
    at Interface.emit (events.js:198:13)
login: login: pass: [ERROR] Get LinkedIn token failed
```

I have confirmed all my passwords I've been entering are correct as well.

**Comment by ChiahungTai** [comment]
Same problem. :(

**Comment by yihong0618** [comment]
@ChiahungTai @zwhitchcox 
I have tried all the methods.
1. LinkedIn seems dead I would try to fix it.
2. GitHub method and Cookie are fine, if you use GitHub method did you make sure your leetcode accout have linked the third-party GitHub.
3. you can user this repo for test https://github.com/leetcode-tools/leetcode-cli

**Comment by yihong0618** [comment]
@ChiahungTai @zwhitchcox
LinkedIn change their login way, and its a bit complicated now, I will try my best to let it work.

**Comment by yihong0618** [comment]
fix: LinkedIn login see https://github.com/leetcode-tools/leetcode-cli/pull/43

**Comment by jdneo** [comment]
SHould be fixed in 0.17.0
